fix isElement function

meinradandermatt opened the pull request:
# Motivation
The isElement function has added a direct dependency to the ownerDocument. This is the reason why for the spellcheck the instanceof will fail. 

# Changelog
🐞 support marker in highlighting with other ownerdocument

jenstroeger commented:
> In Chrome the isElement function is not working as the HTMLElement is not an object there.

Can you please elaborate on “is not working”? What’s your expected and actual result?

I ask because as per [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement): “The `HTMLElement` interface represents any [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML) element” and “[it] inherits properties from its parent, Element.” 

There’s also a difference between [typeof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof) and [instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof) in JavaScript. I think the
```javascript
typeof HTMLElement === 'object'
```
will always be false because `HTMLElement` is a constructor (i.e. function) on Chrome, Firefox, and Safari:
```
> typeof HTMLElement
< "function"
```

marcbachmann commented:
I also think the code should work. Maybe we have an issue in our downstream code that the element isn't from the same window and therefore a different `HTMLElement` instance?

marcbachmann commented:
I'd rather check `typeof obj && obj instanceof obj.ownerDocument?.defaultView?.HTMLElement`

meinradandermatt commented:
> I also think the code should work. Maybe we have an issue in our downstream code that the element isn't from the same window and therefore a different `HTMLElement` instance?
The code work, I just added the update from stackoverflow (where the existing code is coming from) to test. Yes, it is from a different window. As we have added with this code a dependency to the window. So we have added with that a breaking change. I will check what we can do with the spellcheck. 

Just the example which is not working.

`const createElement = (html, win = window) => {
  const el = win.document.createElement('div')
  el.innerHTML = html
  return el.firstElementChild
}

var ifrm = window.document.createElement("iframe")
window.document.body.append(ifrm)

const spanElement = createElement('<span>test</span>', ifrm.contentWindow)
spanElement.setAttribute('data-editable', 'ui-unwrap')
spanElement.setAttribute('data-highlight', 'something')


spanElement instanceof HTMLElement`  // --> false

marcbachmann commented on spec/highlighting.spec.js:
```diff
@@ -111,6 +111,25 @@ describe('Highlighting', function () {
+    it('handle the marker with other ownerdocument correctly', function () {
+      const blockText = 'Mehr als 90 Prozent'
+      const text = 'Mehr als 90 Prozent'
+      const ifrm = window.document.createElement('iframe')
```
`const iframe` would be also ok to use. 😉

meinradandermatt replied:
it's def. too long. 😉

livingdocs-automation commented:
:tada: This PR is included in version 3.0.7 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@livingdocs/editable.js/v/3.0.7)
- [GitHub release](https://github.com/livingdocsIO/editable.js/releases/tag/v3.0.7)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
